corehq/motech/repeaters/models.py

```diff
@@ -163,16 +164,27 @@ def register(self, payload, next_check=None):
         if not self.allowed_to_forward(payload):
             return
 
+        def forward_now(record):
+            from corehq.motech.repeaters.tasks import process_repeat_record
+
+            if not record.redis_lock.acquire(blocking=False):
+                return
+            process_repeat_record.delay(repeat_record)
```

Comment:
I specifically used a different variable name to [avoid it shadowing](http://www.oznetnerd.com/python-shadowing/) `repeat_record` in the outer scope (as you mentioned in Book Club, the reason Python 3 introduced `nonlocal`.) What do you think of using the name "repeat_record_" in the inner scope to make it clear that it's the same thing, but still avoid shadowing the outer variable name? Or do you feel shadowing is a non-issue?

Reply:
My preference here would be to use the same name in both places because it references _the same object_ in both places.

Interesting article. Of the three examples given, I think _Avoid shadowing by using functions_ was my favorite (although I woudln't necessarily employ it in the code we're reviewing here), followed closely by _It’s fine to use shadowing_. I think there are more important best practices, such as writing _short functions_ and having them _do one thing_, that can be followed to make it easier to spot naming errors. The gotcha scenario they outlined where `name` was renamed to `person` didn't really resonate because that's very close to what you were trying to do by applying the _Avoid shadowing by ensuring all names are unique_ rule to `forward_now`, but you still got bitten. In this case you've followed both of the other best practices I mentioned above, and without a doubt that helped me spot the error.

BTW, it still hasn't been fixed. That's my fault for combining two separate issues in the same comment—you fixed the nit, but not the more important issue.